src/components/hooks/useTimezone.ts

```diff
@@ -0,0 +1,55 @@
+import { TimeRange } from '@embeddable.com/core';
+import { addMinutes, subMinutes } from 'date-fns';
+import {
+  parseJSON
+} from 'date-fns';
+
+export function parseTime(dateStr: string): number {
+  const d: Date = parseJSON(dateStr);
+  return d.valueOf() + d.getTimezoneOffset() * 60000;
+}
+
+
+export function timeRangeToUTC(range?: TimeRange) {
+  if (!range?.to || !range?.from) return range;
+
+  return {
+    ...range,
+    to: toUTC(range.to),
```

Comment:
from what I was told, CUBE supports timezones. Do we need to convert the dates to UTC still?

Reply:
we had a discussion with Eyk, i think is no longer need : https://embeddableworkspace.slack.com/archives/C06CYCX4QBF/p1711555831184919

Reply:
@mad-raccoon Tom and I spoke to Misha and understand that Cube supports timezones and we're passing through the info, but it's currently not considering it and therefore isn't working. 

We also did quite a bit of testing of the comparison filter, which uses this, and understand that it does require this function. Once we're correctly using timezones though at the Cube level, I think we'll be able to remove it.